### ISSUE TITLE
Singularity:Overview

### ISSUE THREAD

**Issue by tmozgach** [bcgsc/orca#44]
For Singularity version: `2.4.1`

Singularity Recipe:
https://github.com/bcgsc/orca/blob/master/Singularity.1

Build the image MANUALLY:
```
sudo singularity build orca.simg Singularity1
```

Each user should create an overlay:

```
singularity image.create --size 3000 overlay.simg 
singularity shell --overlay overlay.simg orca.simg
```

Problem: During the building image MANUALLY the following error appears: `Cannot mkdir: No space left on device during building`
Explanation: `/tmp/` is full
Solution: Change cash and tmp directory using environment variables.
`SINGULARITY_CACHEDIR` for persistent things, and `SINGULARITY_TMPDIR` for temporary things during build.

 For example:
```
SINGULARITY_DISABLE_CACHE=Yes
export SINGULARITY_DISABLE_CACHE
SINGULARITY_CACHEDIR=/home/tmozgacheva/tmp/tmp
export SINGULARITY_CACHEDIR
SINGULARITY_TMPDIR=/home/tmozgacheva/tmp
export SINGULARITY_TMPDIR
```

AUTOMATED build in `Singularity hub` (https://singularity-hub.org/) :

Better to set up the manual trigger:
https://github.com/singularityhub/singularityhub.github.io/issues/95

Current issue is not solved:
https://github.com/singularityhub/singularityhub.github.io/issues/94

**Comment by ckeeling** [comment]
I notice your singularity recipe says #PerlIO::gzip fails.  Have you found a solution for this?  This is required for maker, which I have been trying to set up as well, with the same error.

**Comment by ckeeling** [comment]
Hi Shaun,

I'm actually trying to do this within a Singularity recipe, but I'm getting the gcc@4.9 failing before this, so I'm not there yet.  Have you had success getting maker set up?  I'm trying to do this on cedar.

**Comment by sjackman** [comment]
Yes, I have MAKER installed at the GSC using Linuxbrew. Can you use `gcc` rather than `gcc@4.9`?
